src/main/java/org/eclipse/golo/compiler/ir/Decorator.java

```diff
@@ -25,8 +27,13 @@ public ExpressionStatement getExpressionStatement() {
   }
 
   public void setExpressionStatement(ExpressionStatement expr) {
-    if (!(expr instanceof ReferenceLookup || expr instanceof FunctionInvocation)) {
-      throw new IllegalArgumentException("Decorator expression must be a reference of a invocation");
+    if (!(expr instanceof ReferenceLookup
+          || expr instanceof FunctionInvocation
+          || expr instanceof ClosureReference
+          || (expr instanceof BinaryOperation
+            && ((BinaryOperation) expr).getType().equals(OperatorType.ANON_CALL)))) {
```

Comment:
...but here it's more a matter of style than anything else since it's unlikely to be `null`.